src/term.rs

```diff
@@ -1488,6 +1488,15 @@ pub enum NAryOp {
     /// The merge operator in contract mode (see [crate::eval::merge]). The arguments are in order
     /// the contract's label, the value to check, and the contract as a record.
     MergeContract(),
+    /// Seals one record into the tail of another. Used to ensure that functions
+    /// using polymorphic record contracts do not violate parametricity.
+    ///
+    /// Takes four arguments:
+    ///   - a [sealing key](Term::SealingKey), which must be provided later to unseal the tail,
+    ///   - a [label](Term::Lbl), which will be used to assign blame correctly tail access is attempted,
+    ///   - a [record](Term::Record), which is the record we wish to seal the tail into,
+    ///   - the [term](Term) that we wish to seal.
+    SealRecordTail(),
```

Comment:
The same consideration as for `%record_empty_with_tail%`, maybe it would be better to go with `RecordSealTail`?